app/models/pubmed_source_record.rb

```diff
@@ -79,258 +79,8 @@ def pubmed_update
     update! attrs
   end
 
-  def extract_abstract_from_pubmed_record(pubmed_record)
-    pubmed_record.xpath('MedlineCitation/Article/Abstract/AbstractText').text
+  def source_as_hash
+    Pubmed::MapPubHash.map(source_data)
   end
 
```

Comment:
remove all of the pub hash generation related methods to a new class, and call our new class when we ask for the pub hash